### PR TITLE
Extract with subpath

### PR DESCRIPTION
**Pull request by GuillermoDR**
Extract the correct formatting for a passed url including the sub path and change all '/' into '@' (so that it can be in a Firestore str field)

```
>>> extract_with_subpath('http://www.something.com/home/asd.html?abc')
'something.com@home@asd.html'
>>> extract_with_subpath('https://app.example.co.uk/en/about/something.html')
'app.example.co.uk@en@about@something.html'
```

### CODE REVIEW

**Review by saltas888 — APPROVED** [review]
Nice! I'm approving with a couple of minor comments:
1. I'd call the function `extract_with_path` since URL components are these: `scheme://host:port/path?query` (https://tools.ietf.org/html/rfc3986#section-3.3)
2. `@` is a bit confusing with representing an email. What about using backslash(\)?

File: dealroom_urlextract/__init__.py [hunk 1]
```diff
@@ -30,4 +32,29 @@ def extract(url: str):
         else:
             raise Exception('Invalid url format: %s Check url suffix.'%(url))
     else:
-        raise Exception('URL must me string.')
\ No newline at end of file
+        raise Exception('URL must me string.')
```
Comment (vittorio-dealroom) [inline dealroom_urlextract/__init__.py:35]:
```suggestion
        raise Exception('URL must be string.')
```

**Review by vittorio-dealroom — CHANGES_REQUESTED** [review]
super minor change, sorry:P

**Comment by GuillermoDR** [comment]
> Nice! I'm approving with a couple of minor comments:
> 
> 1. I'd call the function `extract_with_path` since URL components are these: `scheme://host:port/path?query` (https://tools.ietf.org/html/rfc3986#section-3.3)
> 2. `@` is a bit confusing with representing an email. What about using backslash(`\`)?

Agree. Renamed the function into `extract_with_path`. Using @ is a standard that we already started using inside Firestore so we have to keep it this way.

**Review by vittorio-dealroom — APPROVED** [review]

**Comment by saltas888** [comment]
> > Nice! I'm approving with a couple of minor comments:
> > 
> > 1. I'd call the function `extract_with_path` since URL components are these: `scheme://host:port/path?query` (https://tools.ietf.org/html/rfc3986#section-3.3)
> > 2. `@` is a bit confusing with representing an email. What about using backslash(`\`)?
> 
> Agree. Renamed the function into `extract_with_path`. Using @ is a standard that we already started using inside Firestore so we have to keep it this way.

That's fair. If it's too late to change it it's fine, but initially I thought it was for extracting email before I read what the method does.

**Pull request merged**